pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/TableConfigUtils.java

```diff
@@ -1002,7 +1002,8 @@ static void validatePartialUpsertStrategies(TableConfig tableConfig, Schema sche
       return;
     }
 
-    Preconditions.checkState(tableConfig.getIndexingConfig().isNullHandlingEnabled(),
+    Preconditions.checkState(schema.isEnableColumnBasedNullHandling()
+            || tableConfig.getIndexingConfig().isNullHandlingEnabled(),
```

Comment:
This is the other change that changes semantic. Before this change partial upserts were not accepted when column based null handling was enabled but table based was disabled.

Reply:
I'm not sure about the impact in partial upsert. I just know the current behavior is incorrect. 

I'm cc'ing @klsince @KKcorps to ask for their feedback